### ISSUE TITLE
strange keyboard layout

### ISSUE THREAD

**Issue by sbarex** [tekezo/Karabiner-Elements#311]
I have a macbook with italian keyboard.
So when I show the keyboard viewer from the menu bar I see this layout:
![schermata 2016-10-15 alle 19 54 18](https://cloud.githubusercontent.com/assets/1948940/19412222/923c9f14-9311-11e6-8efb-7574e754ae56.png)
When I execute the Karabiner element the viewer show a full keyboard with numeric pad.
But this layout don't show the fn button, and also have some keys in different position (note the button on the left of 1, the enter key, the ù, missing the right alt, and some others differences).
![schermata 2016-10-15 alle 19 55 00](https://cloud.githubusercontent.com/assets/1948940/19412248/3b5dd63a-9312-11e6-8c09-ee41fd3e408e.png)

**Comment by mb21** [comment]
I experience the same when running Karabiner-Elements (0.90.50), some additional detail:

I'm using a US keyboard layout on two Swiss (CH-de) hardware keyboards. On the builtin MacBook Air keyboard it works, while on an old external Apple keyboard ([M7803](https://en.wikipedia.org/wiki/Apple_Keyboard#Apple_Pro_Keyboard_.28M7803.29)) I've the problem that the key which usually produces a backtick (`` ` ``, i.e. the one left to the `z` on a US keyboard) doesn't work, producing a `§` instead.

Showing the `Keyboard Viewer` displays the backtick in its proper place (clicking it with the mouse also works), but as soon as I type a character, the keyboard layout shown changes (having no key left of the `z`) and the backtick moves to the key left of the number `1`. Actually pressing the key left to the `1` actually produces a backtick, so I guess I could get used to that...

But what's going on here? Why are the `` ` `` and `§` keys switched on the internal and external keyboards?

**Comment by tekezo** [comment]
Please upgrade Karabiner-Elements to the latest version and set keyboard type in Devices tab.
https://github.com/tekezo/Karabiner-Elements/tree/master/usage#set-keyboard-type

**Comment by mb21** [comment]
Thanks for continuing to improve karabiner!

Setting my keyboard to `ISO` did indeed fix the onscreen "keyboard viewer". However, the `` ` `` and `§` keys are still switched on my external Apple keyboard (M7803) when running Karabiner. To be precise: the keyboard viewer shows all keys where they actually are on my keyboard. However, pressing the key left of the `1` (labelled `§`) on my physical keyboard presses the key left of the `z` in the onscreen keyboard viewer and results in a backtick (and vice versa).

**Comment by jnkvbk** [comment]
I have the same issue. On the mac keyboard The UK PC layout puts | on shift+\, however on my connected Das Keyboard I get a ¬. Pressing \ gives me a ` (backtick).

**Comment by mugukamil** [comment]
I use last version, and I have the same issue on my wired apple keyboard...

**Comment by mrelwood** [comment]
I also have the issue with the Keyboard Viewer switching to Extended. Switching to ISO layout from Karabiner Preferences now seems to give me the proper characters, but the Keyboard Viewer still switches to extended as soon as I hit any key.

Perhaps Karabiner would have to add a laptop keyboard layout to it's Preferences to fix this?

**Comment by paulca55** [comment]
I have the same problem. I use a USB full size Apple keyboard and the physical key layout is the same as that on the MacBook internal keyboard. However when using Karabiner Elements if I set the keyboard type to ISO the external keyboard works fine, but the internal keyboard switches the ` key with the § key. Then if I set the keyboard type to ANSI (which isn't how the internal keyboard looks) the laptop works fine but the external keyboard has those keys switched instead.

It's not exactly a major pain having to switch the keyboard layout when taking the MacBook mobile but it would be great if a setting could be added so I didn't have to do this.

Using version 11.6.0
